packages/server/src/trpc-routers/__tests_e2e__/swap-router.test.ts

```diff
@@ -0,0 +1,1160 @@
+
+  const tokenOutPrice = reply.tokenOutPrice!.toDec().toString();
+  expect(isNumeric(tokenOutPrice)).toBeTruthy();
+  expect(parseFloat(tokenOutPrice)).toBeGreaterThan(0);
+
+  // amount fiat value
+  expect(reply.amountFiatValue).toBeInstanceOf(PricePretty);
+
+  const amountFiatValue = reply.amountFiatValue!.toDec().toString();
+  expect(isNumeric(amountFiatValue)).toBeTruthy();
+  expect(parseFloat(amountFiatValue)).toBeGreaterThan(0);
+});
+
+it("Sidecar - USDT <> USDC - should return valid quote. Token in amount difference should be less than 5% to token out amount", async () => {
+  const tokenInAmount = "1000000";
+  const reply = await caller.swapRouter.routeTokenOutGivenIn({
+    tokenInDenom: usdtMinimalDenom,
+    tokenInAmount,
+    tokenOutDenom: usdcMinimalDenom,
+    preferredRouter: "sidecar",
+  });
+
+  // Amount
+  expect(reply.amount).toBeInstanceOf(CoinPretty);
+  expect(reply.amount.currency).toEqual(makeMinimalAsset(usdcAsset.rawAsset));
+
+  const tokenInAmountDec = new Dec(tokenInAmount).quo(
+    DecUtils.getTenExponentN(usdcAsset.decimals)
+  );
+
+  const amountDec = reply.amount.toDec();
+  const amount = amountDec.toString();
+  expect(isNumeric(amount)).toBeTruthy();
+  // Make sure amount is not negative
+  expect(parseFloat(amount)).toBeGreaterThan(0);
+
+  // Token out amount should be less than 5% of token in amount
+  expect(
+    tokenInAmountDec.sub(amountDec).quo(tokenInAmountDec).lte(new Dec("0.05"))
+  ).toBeTruthy();
+
+  // Swap fee
+  expect(reply.swapFee).toBeInstanceOf(RatePretty);
+  // Should match with the format of "0.1%"
+  expect(reply.swapFee?.toString()).toMatch(percentageRegex);
+
+  const swapFee = reply.swapFee!.toDec().toString();
+  expect(isNumeric(swapFee)).toBeTruthy();
+  expect(parseFloat(swapFee)).toBeGreaterThan(0);
+
+  // Price impact token out
+  expect(reply.priceImpactTokenOut).toBeInstanceOf(RatePretty);
+  // Should match with the format of "0.1%"
+  expect(reply.priceImpactTokenOut?.toString()).toMatch(percentageRegex);
+
+  const priceImpactTokenOut = reply.priceImpactTokenOut!.toDec().toString();
+  expect(isNumeric(priceImpactTokenOut)).toBeTruthy();
+  expect(parseFloat(priceImpactTokenOut)).toBeGreaterThan(0);
+
+  // Token in fee amount
+  expect(reply.tokenInFeeAmount).toBeInstanceOf(Int);
+  const tokenInFeeAmount = reply.tokenInFeeAmount!.toString();
+  expect(isNumeric(tokenInFeeAmount)).toBeTruthy();
+  expect(parseFloat(tokenInFeeAmount)).toBeGreaterThan(0);
+
+  // Split
+  expect(Array.isArray(reply.split)).toBeTruthy();
+  expect(reply.split.length).toBeGreaterThan(0);
+
+  for (const split of reply.split) {
+    expect(split.initialAmount).toBeInstanceOf(Int);
+
+    expect(Array.isArray(split.pools)).toBeTruthy();
+    expect(split.pools.length).toBeGreaterThan(0);
+
+    expect(split.tokenInDenom).toBe(usdtMinimalDenom);
+
+    expect(Array.isArray(split.tokenOutDenoms)).toBeTruthy();
+    expect(split.tokenOutDenoms.length).toBeGreaterThan(0);
+  }
```

Comment:
Are there any repeated checks that we could move into helpers for readability?

Reply:
Thank you for the review! My latest commit abstracts the assertion to a function and it's much cleaner